File: 0060-Permutation Sequence/0060-Permutation Sequence.cpp
class Solution {
public:
    string getPermutation(int n, int k) {
        string result;
        int fac = 1;
        for (int i = 1; i < n; ++i) {
            fac *= i;
        }
        
        vector<char> nums(n);
        for (int i = 1; i <= n; ++i) {
            nums[i - 1] = '0' + i;
        }
        int base = n - 1;
        --k;
        for (int i = 0; i < n; ++i) {
            int index = k / fac;
            k %= fac;
            result += nums[index];
            nums.erase(nums.begin() + index);
            
            if (base) {
                fac /= base;
                --base;
            }
        }
        
        return result;
    }
};

File: 0001-0100/0060-Permutation Sequence/0060-Permutation Sequence.py
import math
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        fac = math.factorial(n - 1)
        nums = [i for i in range(1, 1+n)]
        base = n - 1
        k -= 1
        result = [0] * n
        for i in range(n):
            index = k // fac
            k %= fac
            result[i] = str(nums[index])
            nums.pop(index)
            if base:
                fac //= base
                base -= 1
        return ''.join(result)
